### PR TITLE
[CDAP-9392] DataPrep Workspace Tabs

### PR DESCRIPTION
**Pull request by elfenheart**
JIRA: https://issues.cask.co/browse/CDAP-9392

### CODE REVIEW

File: cdap-ui/app/cdap/components/DataPrep/WorkspaceTabs/index.js [hunk 1]
```diff
@@ -98,129 +96,135 @@ export default class WorkspaceTabs extends Component {
...
-            name="icon-close"
-          />
-        </span>
+          {
+            list.map((workspace) => {
+              return (
+                <div
+                  key={workspace.id}
+                  className="workspace-list-dropdown-item"
+                >
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/DataPrep/WorkspaceTabs/index.js:186]: Would be safer if add a `clearFix` and `float-xs-left` on the other element as we use `float-xs-right` on one of the elements. 

Seeing this in firefox
screen shot 2017-05-24 at 6 43 04 pm

File: cdap-ui/app/cdap/text/text-en.yaml [hunk 2]
```diff
@@ -616,6 +616,13 @@ features:
         uploadBtnLabel: Upload
     Upgrade:
       minimumVersionError: "Data Preparation requires wrangler-service artifact version 1.3.0 or above. Version {highestVersion} found. Please install the latest wrangler-service from Cask Market."
+    WorkspaceTabs:
+      DeleteModal:
+        header: Close {workspace}
+        mainMessage: Are you sure you want to close this tab?
+        helperMessage: Any directives you have applied will be lost.
+        confirmButton: Yes, close tab
+        cancelButton: No, keep tab open
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/text/text-en.yaml]: Alphabetical sorting.

**Review by ajainarayanan — COMMENTED** [review]
Minor comments.

**Comment by ajainarayanan** [comment]
Talked to @elfenheart about two things,
1. Delete confirmation modal - modifying secondary button to link & moving the workspace name to the body of the modal
2. Delete confirmation modal has issues when user agrees to delete the workspace.

**Comment by bdmogal** [comment]
@ajainarayanan the tabs don't collapse when the sidepanel is opened:
image

**Comment by bdmogal** [comment]
THe tabs don't seem to be sorted by label:
image

**Comment by bdmogal** [comment]
Instead of opening this when you click on the < button in DataPrep
image
can we collapse the connections so we see 
image

**Comment by bdmogal** [comment]
The tab caret is misaligned
image

**Comment by bdmogal** [comment]
oh hover, let's make it orange, and remove the underline:
image

Let's make the text of the selected tab orange too:
image

**Comment by bdmogal** [comment]
Let's make the entire area of the trapezium clickable, or at least use a larger icon:
image

**Comment by bdmogal** [comment]
The selected tab should always be shown. It should never be hidden:
image


Maybe the logic is:
1. You always order by tab label, unless there are more tabs than you can fit, so you need the dropdown.
2. If there are more tabs than you can fit, and you select a tab from the dropdown, that tab gets swapped with the last displayed tab.

**Comment by bdmogal** [comment]
@elfenheart can we change this label to File System? 
image

**Comment by bdmogal** [comment]
@elfenheart remove HDFS from here?
image

**Comment by bdmogal** [comment]
@elfenheart the tabs don't collapse when connections are opened, or when the screen size is reduced:
image

**Comment by bdmogal** [comment]
@elfenheart can we fix this - https://github.com/caskdata/cdap/pull/8897#issuecomment-304163554?

**Comment by bdmogal** [comment]
@elfenheart text is getting overlapped: 
image

**Comment by bdmogal** [comment]
@elfenheart the dropdown icon is getting cut:
image

File: cdap-ui/app/cdap/components/DataPrep/WorkspaceTabs/index.js [hunk 3]
```diff
@@ -16,211 +16,251 @@
...
 
-  getWorkspaceList() {
-    getWorkspaceList();
+  splitTabs(workspaceList, maxTabs, activeWorkspaceId = this.state.activeWorkspace) {
+    let displayTabs = workspaceList.slice(0, maxTabs);
+    let dropdownTabs = workspaceList.slice(maxTabs);
+
+    let activeWorkspaceIndex = findIndex(dropdownTabs, { id: activeWorkspaceId });
+
+    console.log('activeWorkspaceIndex', activeWorkspaceIndex);
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/WorkspaceTabs/index.js]: remove

File: cdap-ui/app/cdap/text/text-en.yaml [hunk 4]
```diff
@@ -615,7 +615,13 @@ features:
         uploadBtnLabel: Upload
     Upgrade:
       minimumVersionError: "Data Preparation requires wrangler-service artifact version 2.0.0 or above. Version {highestVersion} found. Please install the latest wrangler-service from Cask Market."
-
+    WorkspaceTabs:
+      DeleteModal:
+        confirmButton: Yes, close tab
+        cancelButton: No, keep tab open
```
Comment (tonybach) [inline cdap-ui/app/cdap/text/text-en.yaml]: nit: let's flip these two lines for correct alphabetical order

**Comment by tonybach** [comment]
Code LGTM 👍

**Comment by bdmogal** [comment]
@elfenheart it seems like you can't close a non-active tab. The tab still stays there. If you navigate to it, it disappears. So I think you can delete it, but we still show it, until the next time you switch tabs. See attached gif 
inactive-tab

**Comment by bdmogal** [comment]
@elfenheart even though there is a lot of space available, the tab goes under the dropdown:
image

**Comment by bdmogal** [comment]
I think the above happens because when uploading the file, I had the collapsed view. When I refresh the page, I can see the tabs correctly without the dropdown.

**Comment by bdmogal** [comment]
tab-space-issue

**Review by bdmogal — APPROVED** [review]

**Pull request merged**